src/java/org/apache/cassandra/config/JMXServerOptions.java

```diff
@@ -0,0 +1,196 @@
+import static org.apache.cassandra.config.CassandraRelevantProperties.JAVAX_NET_SSL_KEYSTORE;
+import static org.apache.cassandra.config.CassandraRelevantProperties.JAVAX_NET_SSL_KEYSTOREPASSWORD;
+import static org.apache.cassandra.config.CassandraRelevantProperties.JAVAX_NET_SSL_TRUSTSTORE;
+import static org.apache.cassandra.config.CassandraRelevantProperties.JAVAX_NET_SSL_TRUSTSTOREPASSWORD;
+import static org.apache.cassandra.config.CassandraRelevantProperties.JAVA_SECURITY_AUTH_LOGIN_CONFIG;
+
+public class JMXServerOptions
+{
+    //jmx server settings
+    public final Boolean enabled;
+    public final Boolean remote;
+    public final int jmx_port;
+    public final int rmi_port;
+    public final Boolean authenticate;
+
+    // ssl options
+    public final EncryptionOptions jmx_encryption_options;
+
+    // options for using Cassandra's own authentication mechanisms
+    public final String login_config_name;
+    public final String login_config_file;
+
+    // location for credentials file if using JVM's file-based authentication
+    public final String password_file;
+    // location of standard access file, if using JVM's file-based access control
+    public final String access_file;
+
+    // classname of authorizer if using a custom authz mechanism. Usually, this will
+    // refer to o.a.c.auth.jmx.AuthorizationProxy which delegates to the IAuthorizer
+    // configured in cassandra.yaml
+    public final String authorizer;
+
+    public JMXServerOptions()
+    {
+        this(null, false, 7199, 0, false,
+             new EncryptionOptions(), null, null, null,
+             null, null);
+    }
+
+    public JMXServerOptions(int jmxPort, boolean local)
+    {
+        this(null, !local, jmxPort, 0, false,
+             new EncryptionOptions(), null, null, null,
+             null, null);
+    }
+
+    public JMXServerOptions(Boolean enabled,
+                            Boolean remote,
+                            int jmxPort,
+                            int rmiPort,
+                            Boolean authenticate,
+                            EncryptionOptions jmx_encryption_options,
+                            String loginConfigName,
+                            String loginConfigFile,
+                            String passwordFile,
+                            String accessFile,
+                            String authorizer)
+    {
+        this.enabled = enabled;
+        this.remote = remote;
+        this.jmx_port = jmxPort;
+        this.rmi_port = rmiPort;
+        this.authenticate = authenticate;
+        this.jmx_encryption_options = jmx_encryption_options;
+        this.login_config_name = loginConfigName;
+        this.login_config_file = loginConfigFile;
+        this.password_file = passwordFile;
+        this.access_file = accessFile;
+        this.authorizer = authorizer;
+    }
+
+    @Override
+    public String toString()
+    {
+        // we are not including encryption options on purpose
```

Comment:
@smiklosovic Can we atleast include the fact if the `jmx_encryption_options` are specified or not and if so if they are enabled (assuming they have different enabled flag compared to `jmx_server_options`.

Reply:
sure